protocol_connect.go

```diff
@@ -511,6 +511,21 @@ func (cc *connectUnaryClientConn) validateResponse(response *http.Response) *Err
 		}
 		cc.responseTrailer[strings.TrimPrefix(k, connectUnaryTrailerPrefix)] = v
 	}
+	err := connectValidateUnaryResponseContentType(
+		cc.marshaler.codec.Name(),
+		cc.duplexCall.Method(),
+		response.StatusCode,
+		response.Status,
+		getHeaderCanonical(response.Header, headerContentType),
+	)
+	if err != nil {
```

Comment:
Can we scope `err` to the `if` block? It's visually a little ugly, but the functions in this portion of the code are long enough that I'd love to minimize scope where possible.

Reply:
I had not done so just because I find it a bit hard to read when the `if` keyword is far from the actual condition. But if that's accepted style, and even preferred in this case, I'll change it.